app/src/main/java/org/dicio/dicio_android/input/InputDevice.java

```diff
@@ -132,6 +140,23 @@ protected void notifyInputReceived(final String input) {
             inputDeviceListener.onInputReceived(input);
         }
     }
+    /**
+     * This has to be called by functions overriding {@link #tryToGetInput(boolean)} when some input
+     * from the user is received. Should be called only once, when stopping to get input. Should not
+     * be called if {@link #notifyNoInputReceived()} is called instead.
```

Comment:
Also change this for `notifyInputReceived(String)` 
```suggestion
     * be called if {@link #notifyNoInputReceived()} or {@link #notifyInputReceived(String)} are called instead.
```